### ISSUE TITLE
[Geofence] Exit event will be invoked twice in the latest version

### ISSUE THREAD

**Issue by xinmeng1** [domaven/xamarin-plugins#11]
I encounter one problem when using the latest version. That is the Exit event will be invoked twice. It means that Enter the region occur one time, however the Exited the region occur 2 times. I have try the sample which use the 1.0.4 without this problem. However I update the plugin to 1.1.1, the problem occurred again in the Sample. I think it is a bug maybe.

Test Case: Using iOS Emulator

Set the location as the custom location:

La: 51.625742
Lo: -0.7424

Open the Sample App

Add the fence at current location and radius is 10 

Then choose the City Run in the Location emulator.

There will be two notifications for Exited geofence region.

![skitch](https://cloud.githubusercontent.com/assets/3052251/14789793/198e27ce-0b06-11e6-8934-ef29841464e1.png)

**Comment by jfversluis** [comment]
+1 but not just twice. It seems to fire randomly at other places as well.
Enter event is just being fired once.

**Comment by klingu** [comment]
I have 2 issues/bugs for your plugin: which I can not explain after analysing your sample app and my own app.

1: double exitnotifications on exiting geofence.
2: On add geofence, and GPS location is outside of the added geofence. exitnotification is shown.

**AD 1**
I run the sample app and my own app with the same version of your Geofence plugin (1.1.2) This issue is also on version (1.1.1).

I've added 2 locations in the sample app... (Start) and (kruis1)
When entering these regions, the EnterRegionMessage appears once for each region.
When exiting these regions, the ExitRegionMessage appears just once for each region.

In my own application, I've added 1 location (Koningskade 4)
On entering this region, the EnterRegionMessage appears just once.
On exiting this region, the ExitRegionMessage appears twice.
![simulator screen shot 08 jul 2016 14 13 01](https://cloud.githubusercontent.com/assets/7271479/16686981/25fecf6e-4516-11e6-978e-958151d8cb43.png)

**AD2**
When I add a location via an Async Task, just like the same way you do, in my own app: I immediately get the notification I've exited the geofence. 

When adding a location on the sample app, it does not show any ExitRegionMessage at all.

I'm willing to share my Solution with you.

Like to hear from you soon,

Jeffrey
